src/PopChainManager/index.ts

```diff
@@ -8,6 +8,35 @@ export class PopChainManager {
     return closest(pop.targetEl, '[pop]', true);
   }
 
+  public getFullPopChain(pop: Pop, hideFullChain: boolean): Pop[] {
```

Comment:
Not exactly equivalent, but I implemented an iterative solution based on yours! Let me know what you think! :)